standard-bridge-custom-token/contracts/L2CustomERC20.sol

```diff
@@ -10,6 +10,7 @@ contract L2CustomERC20 is L2StandardERC20 {
     )
         L2StandardERC20(_l2Bridge, _l1Token, "Custom L2 Token", "L2T")
         {
-            _setupDecimals(8);
+            // _setupDecimals(8);
+            _mint(msg.sender, 10000);
```

Comment:
Why has the logic we use here changed? If `setupDecimals` is a missing method we should just fix that instead of changing it. Otherwise minting is explicitly disallowed on L2 and thus this is not a suitable example. Otherwise we'll have a token with supply on L2 > supply on L1 thus able to drain Bridge of deposited tokens.

Reply:
`_setupDecimals` is indeed missing in the latest Open Zeppelin package, and because I didn't know why I didn't want to mess with it.

However, I like having `_mint` here precisely because it is a bad idea. This means that I can sneak in the extra point that they should not do it (see lines 18-20 of the README.md, but I should also put it in the code itself as a comment). It's better training to show something stupid, and explain why it is stupid, than to ignore the issue.

Reply:
If there's going to be a tutorial containing anti-practices it needs to be separated from this which purpose it is to give a template to people to work _from_ not _against_. For example see the custom token for WBTC [here](https://github.com/ethereum-optimism/standard-bridge-tokens/pull/2/files#diff-9e730f4d0756b03cf9d2c3f1f23d216c2486810a03489649728cc449a035bf94). This tutorial was written to be used as a "clone this repo and switch the params to deploy your token" not a "look at this and avoid doing this in your implementation".

Reply:
I said what you asked. I still think we're missing an opportunity by not teaching this, considering that the custom ERC20 code is the one part of the tutorial that students will **need** to change (if they don't need special functionality they wouldn't be using this tutorial), but regenesis is Thursday. I decided it's more important to have tutorials available immediately after regenesis than to argue about how to teach things.